src/Disks/TemporaryFileOnDisk.cpp

```diff
@@ -41,17 +40,12 @@ TemporaryFileOnDisk::TemporaryFileOnDisk(const DiskPtr & disk_, const String & p
 
     ProfileEvents::increment(ProfileEvents::ExternalProcessingFilesTotal);
 
-    /// Do not use default temporaty root path `/tmp/tmpXXXXXX`.
-    /// The `dummy_prefix` is used to know what to replace with the real prefix.
-    String dummy_prefix = "a/";
-    relative_path = Poco::TemporaryFile::tempName(dummy_prefix);
-    dummy_prefix += "tmp";
-    /// a/tmpXXXXX -> <prefix>XXXXX
-    assert(relative_path.starts_with(dummy_prefix));
-    relative_path.replace(0, dummy_prefix.length(), prefix);
-
-    if (relative_path.empty())
-        throw Exception(ErrorCodes::LOGICAL_ERROR, "Temporary file name is empty");
+    /// A disk can be remote and shared between multiple replicas.
+    /// That's why we shouldn't use Poco::TemporaryFile::tempName() here (a process id can be the same for different processes on different nodes).
+    static Poco::UUIDGenerator generator;
+    UUID id;
+    generator.createRandom().copyTo(reinterpret_cast<char *>(&id));
+    relative_path = prefix + toString(id);
```

Comment:
`Poco::TemporaryFile::tempName()` generated something like

```
<process_id><six_letters>
```
for example `57aaaaaa, 57aaaaab, 57aaaaac, ...`

It worked well for a local disk, but for a remote disk process id is not unique anymore. UUIDs should work better.